easybuild/tools/modules.py

```diff
@@ -1076,6 +1088,14 @@ def remove_module_path(self, path, set_mod_paths=True):
             self.set_mod_paths()
 
 
+class EnvironmentModules(EnvironmentModulesTcl):
+    """Interface to environment modules 4.0+"""
+    COMMAND = os.path.join(os.environ['MODULESHOME'], 'libexec', 'modulecmd.tcl')
```

Comment:
This should be done more carefully, if `$MODULESHOME` is not defined, this will lead to a nasty crash/traceback.

My suggestion is to use a fallback value, like this:

```python
COMMAND = os.path.join(os.getenv('MODULESHOME', 'MODULESHOME_NOT_DEFINED'), 'libexec', 'modulecmd.tcl')

Reply:
DO we need both MODULESHOME and MODULESHOME_NOT_DEFINED? If using os.getenv and MODULESHOME is undefined, None will be returned and the traceback won't occur. In that case, eb would look for None/libexec/modulecmd.tcl and throw an exception.

Reply:
@ebagrenrut Yes, we need the fallback value, since `os.path.join` won't be happy with the `None` value:

```
>>> os.path.join(None, 'libexec', 'modulecmd.tcl')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/posixpath.py", line 70, in join
    elif path == '' or path.endswith('/'):
AttributeError: 'NoneType' object has no attribute 'endswith'
```